mlflow/utils/search_utils.py

```diff
@@ -187,7 +198,9 @@ def _invalid_statement_token(cls, token):
     def _process_statement(cls, statement):
         # check validity
         invalids = list(filter(cls._invalid_statement_token, statement.tokens))
-        if len(invalids) > 0:
+        if len(invalids) == 3 and invalids[1].value.lower() in ['like', 'ilike']:
```

Comment:
@t-henri This logic doesn't seem to support queries with multiple clauses where one or more clauses contains an `ilike`. For example, attempting the following query from the MlflowClient fails at the parsing stage:
```
> client.search_runs(experiment_ids="0", filter_string="tags.tag_4 = '4' and tags.tag_3 ilike '%3'")
> INVALID_PARAMETER_VALUE: Invalid comparison clause. Expected 3 tokens found 5
```

In contrast, `client.search(experiment_ids=0, filter_string="tags.tag_4 = '4' and tags.tag_3 = '3'")` succeeds.

Reply:
Ho I'm sorry I didn't catch that, I updated my PR with a fix (and added some tests with multiple conditions) but I'd like your opinion on the implementation. 
The issue is that sqlparse doesn't create Comparison objects when the comparison string has a like/ilike operator, so I had to create it myself and it looks a bit hacky. What do you think about that ? It works but if you have a better/more elegant solution feel free to ping me.